client_code/forms/routes/ballot/Read/__init__.py

```diff
@@ -13,9 +13,21 @@ def __init__(self, ballot, **properties):
             f"https://blockscout.scroll.io/address/{manager.address}"
         )
         self.init_components(**properties)
+        self.update_status()
 
     def hide_button_click(self, **event_args):
         anvil.get_open_form().hide_detail()
 
     def vote_button_click(self, **event_args):
         proof.prove_eligibility(self.ballot)
+
+    def update_status(self):
```

Comment:
sweep: this would be better as a property on the Ballot class with the only logic in the form being for the text and colors to display as a result.